### PR TITLE
test for get_array_metadata

### PR DESCRIPTION
**Pull request by oskooi**
Adds a Python test for `get_array_metadata` involving `symmetries` and `k_point` following #811 which was recently merged.

### CODE REVIEW

File: python/tests/array_metadata.py [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+                            boundary_layers=[mp.PML(dpml)])
+
+        dft_obj = sim.add_dft_fields([mp.Ez], fcen, fcen, 1, where=nonpml_vol)
+        sim.run(until_after_sources=100)
+
+        Ez = sim.get_dft_array(dft_obj,mp.Ez,0)
+        (X,Y,Z,W) = sim.get_array_metadata(dft_cell=dft_obj)
+        Eps = sim.get_array(vol=nonpml_vol,component=mp.Dielectric)
+        EpsE2 = np.real(Eps*np.conj(Ez)*Ez)
+        pulse_modal_volume = np.sum(W*EpsE2)/np.max(EpsE2)
```
Comment (stevengj) [inline python/tests/array_metadata.py]: You might want to instead use some test integrand that involves (x,y,z) as well, e.g. the same integral but put a factor of (x^2 + 2y^2 + 3z^2) in the numerator.

For comparison, compute the same integral using `integrate_field_function` in the `ContinuousSource` simulation.

**Comment by oskooi** [comment]
The test has been updated. It seems that there is a similar test [`modal_volume_metadata.py`](https://github.com/NanoComp/meep/blob/master/python/tests/modal_volume_metadata.py) that was previously added by @HomerReid but which is never actually used since it is not included in [`python/Makefile.am`](https://github.com/NanoComp/meep/blob/master/python/Makefile.am). Perhaps this file was used for internal development and thus should be removed?

File: python/tests/array_metadata.py [hunk 2]
```diff
@@ -14,49 +14,64 @@ def test_array_metadata(self):
...
-                            boundary_layers=[mp.PML(dpml)])
+                            symmetries=symmetries,
+                            boundary_layers=pml_layers)
 
         sim.init_sim()
         sim.solve_cw(1e-6, 1000, 10)
-        cw_modal_volume = sim.modal_volume_in_box(box=nonpml_vol)
+
+        def electric_energy(r, ex, ey, ez, eps):
+            return np.real(eps * (np.conj(ex)*ex + np.conj(ey)*ey + np.conj(ez)*ez))
```
Comment (stevengj) [inline python/tests/array_metadata.py]: To make sure the `X`, `Y`, `Z` arrays are correct, see my suggestion below about multiplying by a function of `r` too.

(We don't need to test this on a physically meaningful integrand.)

**Comment by oskooi** [comment]
Following #831, the test has been updated to include an arbitrary position-dependent function in the integrand.

**Pull request merged**